### PR TITLE
feat: Masterable attribute

### PR DESCRIPTION
**Pull request by SnowyLeopard**
### What did you fix? 
Adds a new attribute to items `masterable` denoting whether the item is masterable or not.
Certain categories contain masterable items, these are listed in a new config file `masterableCategories`.
When an item matches the category it will be listed as masterable.

---

### Reproduction steps

---

### Considerations
- Does this contain a new dependency? **[No]**
- Does this introduce opinionated data formatting or manual data entry? **[Yes]**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **[Yes]**
- Have I run the linter? **[Yes]**
- Is is a bug fix, feature request, or enhancement? **[Feature]**

### CODE REVIEW

**Comment by TobiTenno** [comment]
woot, ty, bud! not trying to be a pain, it's just harder to review with that in :D

File: build/parser.mjs [hunk 1]
```diff
@@ -188,6 +189,7 @@ class Parser {
     this.addResistanceData(result, category);
     this.addRelics(result, data.relics);
     this.applyOverrides(result);
+    this.applyMasterable(result);
```
Comment (TobiTenno) [inline build/parser.mjs:192]: this should come before `this.applyOverrides` so that the overrides apply last

File: index.d.ts [hunk 2]
```diff
@@ -52,6 +52,7 @@ declare module 'warframe-items' {
         description?: string;
         releaseDate?: string;
         excludeFromCodex?: boolean;
+        masterable?: boolean;
```
Comment (TobiTenno) [inline index.d.ts:55]: given that it's applying on all items, shouldn't be optional, like it should fail types if one of the items is missing this altogether...

```suggestion
        masterable: boolean;
```

File: test/index.spec.mjs [hunk 3]
```diff
@@ -2,6 +2,9 @@ import assert from 'node:assert';
 import { resolve } from 'node:path';
 import { createRequire } from 'module';
 import dedupe from '../build/dedupe.mjs';
+import readJson from '../build/readJson.mjs';
+
+const masterableCategories = await readJson(new URL('../config/masterableCategories.json', import.meta.url));
```
Comment (TobiTenno) [inline test/index.spec.mjs:7]: we already have a bypass as part of the main load for doing that, should just use the require we already have, as that will read the json... (i.e. don't need `readJson`)

**Review by TobiTenno — CHANGES_REQUESTED** [review]
i like the change overall, just some nit-picks

**Comment by SnowyLeopard** [comment]
> woot, ty, bud! not trying to be a pain, it's just harder to review with that in :D

Haha no problem! I completely understand ;)
I've processed the feedback you had, thanks for that! Makes the code better 😄

**Pull request merged**